### ISSUE TITLE
Potential source breaking changes due to params collection

### ISSUE THREAD

**Issue by hez2010** [dotnet/roslyn#73857]
**Version Used**: 9.0.100-preview.6.24304.2

**Steps to Reproduce**:

```cs
string.Concat(["a", "b"]);
```

**Diagnostic Id**: CS0121

**Expected Behavior**: No compilation error

**Actual Behavior**:
```
CS0121 The call is ambiguous between the following methods or properties: 'string.Concat(scoped ReadOnlySpan<object?>)' and 'string.Concat(scoped ReadOnlySpan<string?>)'
```

Note that this is a source breaking change from C# 12 where the code used to compile.

**Comment by CyrusNajmabadi** [comment]
@cston I believe this is the issue you were mentioning to me.   We should see what the spec says and if we got it wrong in the spec or the impl.  My expectation would 100% be that `ROS<T>` is better than `ROS<U>` when there is an implicit conversion from T to U.  Just like `void X(T t)` is better than `void X(U u)` when dealing with a single element.

I'm thinking we might have spec'ed this wrong, but the above was definitely the intent.  :)

**Comment by cston** [comment]
> My expectation would 100% be that `ROS<T>` is better than `ROS<U>` when there is an implicit conversion from T to U.

Currently, [*better conversion*](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#overload-resolution) rules for collection expressions treat this case as ambiguous.

As you suggested, perhaps there should be an additional rule to prefer `System.ReadOnlySpan<E₁>` over `System.ReadOnlySpan<E₂>` if an implicit conversion exists from `E₁` to `E₂`.

**Comment by jaredpar** [comment]
Note that this is a `net9.0` only problem, or at least primarily. If we do take changes here to betterness think that we need to consider typing it to langversion 13
